src/decoder.rs

```diff
@@ -120,6 +120,36 @@ pub fn strip_junk_header(slice: &[u8]) -> io::Result<&[u8]> {
     Ok(&slice[slice.len()..])
 }
 
+/// Decodes range mappping bitfield string into index
+fn decode_rmi(rmi_str: &str) -> Result<u32> {
+    let mut val = 0u32;
+
+    for &byte in rmi_str.as_bytes() {
+        // A: 0b000000
+        // B: 0b000001
+        // g: 0b100000
+        if byte == b'A' {
+            val += 6;
+            continue;
+        }
```

Comment:
I don’t think this is correct.
In each loop iteration, you would need to shift the result value.

Assuming that the bitfield encoding is ordered left to right, you need to shift by `6` bits.

So that `ABg` is interpreted as:
```
  0b000000 << 12 |
        0b000001 << 6 |
              0b100000
= 0b000000000001100000
```

Reply:
@sokra Is this correct?

AAB is then

```
   0b000000 << 12 |
                0b000000 << 6 |
                              0b000001
= 0b000000000000000001
```

so it's  `0b000000 0b000000 0b000001`, but I'm not sure how can `0x00 0x00 0x01` can encode 13

Reply:
Sorry, it was my misunderstanding